File: app/src/main/java/cz/kutner/comicsdb/connector/converter/ClassifiedConverter.java
package cz.kutner.comicsdb.connector.converter;

import org.jsoup.Jsoup;
import org.jsoup.nodes.Document;
import org.jsoup.nodes.Element;

import java.lang.reflect.Type;
import java.util.ArrayList;
import java.util.List;

import cz.kutner.comicsdb.Utils;
import cz.kutner.comicsdb.model.Classified;
import retrofit.converter.ConversionException;
import retrofit.converter.Converter;
import retrofit.mime.TypedInput;
import retrofit.mime.TypedOutput;

public class ClassifiedConverter implements Converter {
    @Override
    public Object fromBody(TypedInput body, Type type) throws ConversionException {
        List<Classified> result = new ArrayList<>();
        Document doc;
        try {
            doc = Jsoup.parse(body.in(), "windows-1250", "");
            for (Element entry : doc.select("div#prispevek")) {
                String nick = entry.select("span.prispevek-nick").get(0).text();
                String category = entry.select("span.prispevek-nick").get(1).text();
                String time = entry.select("span.prispevek-cas").get(0).text();
                String iconUrl = entry.select("div#prispevek-icon").select("img").first().attr("src");
                for (Element remove : entry.select("span,img")) {
                    remove.remove();
                }
                String text = entry.select("div#prispevek-text").html().replace("| ", "").replace("<br></br>", "");
                Classified classified = new Classified(nick, time, category, text);
                if (!iconUrl.isEmpty()) {
                    classified.setIconUrl(Utils.fixUrl(iconUrl));
                }
                result.add(classified);
            }
        } catch (Exception e) {
        }
        return result;
    }

    @Override
    public TypedOutput toBody(Object object) {
        return null;
    }
}

File: app/src/main/kotlin/cz/kutner/comicsdb/connector/converter/ClassifiedConverter.kt
package cz.kutner.comicsdb.connector.converter

import cz.kutner.comicsdb.utils.Utils
import cz.kutner.comicsdb.model.Classified
import org.jsoup.Jsoup
import org.jsoup.nodes.Document
import retrofit.converter.ConversionException
import retrofit.converter.Converter
import retrofit.mime.TypedInput
import retrofit.mime.TypedOutput
import java.lang.reflect.Type
import java.util.*

class ClassifiedConverter : Converter {
    @Throws(ConversionException::class)
    override fun fromBody(body: TypedInput, type: Type): Any {
        val result = ArrayList<Classified>()
        val doc: Document
        try {
            doc = Jsoup.parse(body.`in`(), "windows-1250", "")
            for (entry in doc.select("div#prispevek")) {
                val nick = entry.select("span.prispevek-nick")[0].text()
                val category = entry.select("span.prispevek-nick")[1].text()
                val time = entry.select("span.prispevek-cas")[0].text()
                val iconUrl = entry.select("div#prispevek-icon").select("img").first().attr("src")
                for (remove in entry.select("span,img")) {
                    remove.remove()
                }
                val text = entry.select("div#prispevek-text").html().replace("| ", "").replace("<br></br>", "")
                val classified = Classified(nick, time, category, text)
                if (!iconUrl.isEmpty()) {
                    classified.iconUrl = Utils.fixUrl(iconUrl)
                }
                result.add(classified)
            }
        } catch (e: Exception) {
        }

        return result
    }

    override fun toBody(`object`: Any): TypedOutput? {
        return null
    }
}
